include/nlohmann/json_fwd.hpp

```diff
@@ -70,6 +71,20 @@ struct ordered_map;
 /// @sa https://json.nlohmann.me/api/ordered_json/
 using ordered_json = basic_json<nlohmann::ordered_map>;
 
+/// @brief a minimal specialization that uses the base class json_base_class_with_start_end_markers
+using json_with_start_end_markers = nlohmann::basic_json <
```

Comment:
Open to discussion for names :). I'm terrible at coming up with good names

Reply:
How about `json_base_with_positions`? or `json_base_positional`?

Reply:
Implemented the macros, so base class is gone. basic_json now exposes two property getters: `start_pos()` and `end_pos()`, if the `DIAGNOSTIC_POSITIONS` macro is set. Otherwise there is no additional memory footprint.

Reply:
For this PR, changing the name to `JSON_DIAGNOSTIC_POSITION` would be sufficient. As possible next step, we should also pick up the positions to provide better error message since we now know the positions of each part of the JSON.